### PR TITLE
Make SHS consumers clear in demand plot

### LINKED ISSUE #89: Make consumers included in SHS more obvious in demand view
- One idea: Making the filled areas in the plot patterned https://plotly.com/python/pattern-hatching-texture/ and indicate in some way that the patterned timeseries are not included in the minigrid
- Alternatively or additionally: Display the rows with the SHS timeseries with a different color in some way (border or background of the form)

### PR DESCRIPTION
**Pull request by paulapreuss**
closes #89. SHS consumers are now plotted with hatching and grouped in the plot. They can all be deselected from the plot at once when the group is clicked. 

This PR also fixes a bug that was sometimes targetting the wrong consumer group in the plot when deleting a group.

### CODE REVIEW

File: app/templates/cp_nigeria/steps/scenario_demand.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 
 {% block start_body_scripts %}
-<script src="https://cdn.plot.ly/plotly-latest.min.js"></script>
```
Comment (Bachibouzouk) [inline app/templates/cp_nigeria/steps/scenario_demand.html:8]: latest is not the latest version?

File: app/static/js/update_demand_graph.js [hunk 2]
```diff
@@ -1,19 +1,60 @@
 /*jshint esversion: 6 */
 // variable to store the timeseries as they are retrieved from the database
 var timeseriesData = {};
+var householdTraces = {};
```
Comment (Bachibouzouk) [inline app/static/js/update_demand_graph.js:4]: I am not sure how this variable gets updated? In the code below it seems to me it is only updated if it is not empty

File: app/templates/cp_nigeria/steps/scenario_demand.html (hunk 1 above)
Comment (paulapreuss) [reply app/templates/cp_nigeria/steps/scenario_demand.html:8]: for some reason it doesn't seem like it. I found someone else wondering the same thing https://community.plotly.com/t/why-is-plotly-latest-still-1-58-5/59835 but it is unanswered

File: app/static/js/update_demand_graph.js (hunk 2 above)
Comment (Bachibouzouk) [reply app/static/js/update_demand_graph.js:4]: Nevermind I overlooked `householdTiers` for `householdTraces`

**Review by Bachibouzouk — APPROVED** [review]
Hi Paula,

Nice job, I tested it and could not make it crash, I updated the online tool already with this branch.

I just have a comment about the plot traces:

The "served by SHS" traces are grouped together but this is not the case for the other traces, the other household traces are randomly appearing in the trace list. It could be nice to also group them (the problem if we do this with legend group, is that one cannot toggle only one trace of the group, the entire group is toggled)

**Pull request merged**